pkg/api/validation/validation.go

```diff
@@ -2865,6 +2865,22 @@ func ValidateService(service *api.Service) field.ErrorList {
 		nodePorts[key] = true
 	}
 
+	// Check for duplicate Ports, considering (protocol,port) pairs
+	portsPath = specPath.Child("ports")
+	ports := make(map[api.ServicePort]bool)
+	for i, port := range service.Spec.Ports {
+		if port.Port == 0 {
+			continue
+		}
+		portPath := portsPath.Index(i)
+		key := api.ServicePort{Protocol: port.Protocol, Port: port.Port}
+		_, found := ports[key]
+		if found {
+			allErrs = append(allErrs, field.Duplicate(portPath.Child("port"), port.Port))
```

Comment:
I'm not sure two errors make sense either... a single error that included both the protocol and the port number is what I would expect - `field.Duplicate(portPath, key)`